app/component/itinerary/navigator/hooks/useRealtimeLegs.js

```diff
@@ -149,7 +172,27 @@ const useRealtimeLegs = (initialLegs, mapRef, relayEnvironment) => {
     return () => clearInterval(interval);
   }, [fetchAndSetRealtimeLegs]);
 
-  return { realTimeLegs, time };
+  let currentIndex = -1;
+
+  if (realTimeLegs) {
+    if (time >= legTime(realTimeLegs[realTimeLegs.length - 1].end)) {
+      currentIndex = realTimeLegs.length - 1;
+    } else {
+      currentIndex = realTimeLegs?.findIndex(
+        l => legTime(l.start) <= time && time <= legTime(l.end),
```

Comment:
Note that currentIndex might remain undefined.

Reply:
That's an oversight by me; good catch! will fix with a default `-1` value